tests/layer_tests/tensorflow_tests/test_tf_TensorScatterUpdate.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import platform
+import pytest
+import tensorflow as tf
+from common.tf_layer_test_class import CommonTFLayerTest
+
+rng = np.random.default_rng(872173)
+
+
+class TestTensorScatterUpdate(CommonTFLayerTest):
+    def _prepare_input(self, inputs_info):
+        assert 'tensor:0' in inputs_info
+        assert 'indices:0' in inputs_info
+        assert 'updates:0' in inputs_info
+
+        tensor_shape = inputs_info['tensor:0']
+        updates_shape = inputs_info['updates:0']
+        indices_shape = inputs_info['indices:0']
+
+        inputs_data = {}
+        if np.issubdtype(self.data_type, np.floating):
+            inputs_data['tensor:0'] = rng.uniform(-5.0, 5.0, tensor_shape).astype(self.data_type)
+            inputs_data['updates:0'] = rng.uniform(-5.0, 5.0, updates_shape).astype(self.data_type)
+        elif np.issubdtype(self.data_type, np.signedinteger):
+            inputs_data['tensor:0'] = rng.integers(-8, 8, tensor_shape).astype(self.data_type)
+            inputs_data['updates:0'] = rng.integers(-8, 8, updates_shape).astype(self.data_type)
+        else:
+            inputs_data['tensor:0'] = rng.integers(0, 8, tensor_shape).astype(self.data_type)
+            inputs_data['updates:0'] = rng.integers(0, 8, updates_shape).astype(self.data_type)
+        
+        tensor_rows = tensor_shape[0]
+        tensor_cols = tensor_shape[1]
+
+        indices_rows, indices_cols = indices_shape
+
+        all_indices = []
+        if indices_cols == 1:
+            for row_ind in range(0, tensor_rows):
+                all_indices.append([row_ind])
+        else:
+            for row_ind in range(0, tensor_rows):
+                for col_ind in range(0, tensor_cols):
+                    all_indices.append([row_ind, col_ind])
```

Comment:
Now you implement for 1D and 2D indices. Please generalize method of generation all element indices for given shape.
Here is an idea using `arange, meshgrid and stack`
![image](https://github.com/user-attachments/assets/e7c9cc66-98a9-4648-a2b0-0bcc3e333c68)

So you don't need any if-else construction etc.

Reply:
Thanks for the hint! I think I got it now.